src/raftstore/store/worker/apply.rs

```diff
@@ -248,17 +257,17 @@ struct ApplyContextCore<'a> {
     // `sync_log_hint` indicates whether synchronize wal is prefered.
     sync_log_hint: bool,
     exec_ctx: Option<ExecContext>,
-    use_delete_range: bool,
 }
 
-impl<'a> ApplyContextCore<'a> {
-    pub fn new(host: &'a CoprocessorHost, importer: &'a SSTImporter) -> ApplyContextCore<'a> {
-        ApplyContextCore {
+impl ApplyContext {
+    pub fn new(host: Arc<CoprocessorHost>, importer: Arc<SSTImporter>) -> ApplyContext {
```

Comment:
You can just use `Self` here instead of `ApplyContext`. Same with the line below.